wrapanapi/systems/nuage.py

```diff
@@ -71,6 +78,39 @@ def list_enterprises(self):
     def list_domains(self):
         return self._request_list('/domains', 'get')
 
+    def list_l2_domains(self):
+        return self._request_list('/l2domains', 'get')
+
+    def list_floating_ips(self):
+        return self._request_list('/floatingips', 'get')
+
+    def list_vports_for_domain(self, domain_id):
+        return self._request_list('/domains/{}/vports'.format(domain_id), 'get')
+
+    def list_vports_for_l2_domain(self, l2_domain_id):
+        return self._request_list('/l2domains/{}/vports'.format(l2_domain_id), 'get')
+
+    def list_cloud_subnets(self):
+        return self.list_subnets() + self.list_l2_domains()
+
+    def list_vports(self):
+        res = []
+        for domain in self.list_domains():
+            res += self.list_vports_for_domain(domain['ID'])
+        for l2_domain in self.list_l2_domains():
+            res += self.list_vports_for_l2_domain(l2_domain['ID'])
+        return res
+
+    def list_shared_network_resources(self, type=None):
+        return [
+            res for res in self._request_list('/sharednetworkresources', 'get') if
+            res['type'] == type or type is None
```

Comment:
Please use a different kwarg from type so as not to overload python's `type` - makes this statement a bit confusing

Reply:
Done.